server/src/main/java/org/elasticsearch/index/engine/ElasticsearchConcurrentMergeScheduler.java

```diff
@@ -66,11 +69,14 @@
     private final Set<OnGoingMerge> onGoingMerges = ConcurrentCollections.newConcurrentSet();
     private final Set<OnGoingMerge> readOnlyOnGoingMerges = Collections.unmodifiableSet(onGoingMerges);
     private final MergeSchedulerConfig config;
+    private volatile long lastMergeMillis;
```

Comment:
I don't think we need these changes. I guess it would be enough to check `Engine#getMergeStats()` and then do:

`mergeStats.current > 0 || mergeStats.total != previousStats.total`?